lib/internal/modules/esm/package_config.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const {
+  JSONParse,
+  SafeMap,
+  StringPrototypeEndsWith,
+} = primordials;
+const { URL, fileURLToPath } = require('internal/url');
+const {
+  ERR_INVALID_PACKAGE_CONFIG,
+} = require('internal/errors').codes;
+
+const packageJsonReader = require('internal/modules/package_json_reader');
+
+
+/**
+ * @typedef {string | string[] | Record<string, unknown>} Exports
+ * @typedef {'module' | 'commonjs'} PackageType
+ * @typedef {{
+ *   pjsonPath: string,
+ *   exports?: ExportConfig,
+ *   name?: string,
+ *   main?: string,
+ *   type?: PackageType,
+ * }} PackageConfig
+ */
+
+const packageJSONCache = new SafeMap(); /* string -> PackageConfig */
+
+
+/**
+ * @param {string} path
+ * @param {string} specifier
+ * @param {string | URL | undefined} base
+ * @returns {PackageConfig}
+ */
+ function getPackageConfig(path, specifier, base) {
+  const existing = packageJSONCache.get(path);
+  if (existing !== undefined) {
+    return existing;
+  }
+  const source = packageJsonReader.read(path).string;
+  if (source === undefined) {
+    const packageConfig = {
+      pjsonPath: path,
+      exists: false,
+      main: undefined,
+      name: undefined,
+      type: 'none',
+      exports: undefined,
+      imports: undefined,
+    };
+    packageJSONCache.set(path, packageConfig);
+    return packageConfig;
+  }
+
+  let packageJSON;
+  try {
+    packageJSON = JSONParse(source);
+  } catch (error) {
+    throw new ERR_INVALID_PACKAGE_CONFIG(
+      path,
+      (base ? `"${specifier}" from ` : '') + fileURLToPath(base || specifier),
+      error.message
+    );
+  }
+
+  let { imports, main, name, type } = packageJSON;
+  const { exports } = packageJSON;
+  if (typeof imports !== 'object' || imports === null) { imports = undefined };
```

Comment:
```suggestion
  if (typeof imports !== 'object' || imports === null) {
    imports = undefined;
  }
```

Same suggestion for the other if blocks